### DISCUSSION TITLE
Is possible to create contact forms?

### DISCUSSION THREAD

**Discussion by ozeberle** [directus/directus#2719] [Archive]
Hi guys, is there any out-of-the-box way to create a contact form from Directus?
What I want is to create a subscribe form (form, input text, submit button) which will be implemented in my front web page.
Also, the idea in the near future is to allow the users to create a custom form.

**Comment by rijkvanzanten** [comment]
Heya! Seeing Directus is a headless CMS, there's no "contact form generator" or something like that. Instead, you'd simply create a collection for your submissions, give write access to the public role, and POST your form submissions to the `/items/<submissions collection>` endpoint 👍

**Reply by tulindor** [reply]
Hi, there is an example of this anywhere? i need to send a contact form from a  react app, using directus as CMS.

**Reply by cupcakearmy** [reply]
Well basically you would simply create a collection with the fields you need, and the either make it publicly writable or if you are working in a JAM environment you can simply deploy a cloud function with recapcha and a private API token that actually writes the submissions to protect yourself from spam

**Reply by sgreinerCNS** [reply]
I dont see a way to spam protect this public writeable collection?

**Reply by JamesW1** [reply]
@sgreinerCNS You can enable the built-in rate limiter (`RATE_LIMITER_ENABLED=true`) to limit hits per IP, though for fuller spam protection you'd want to add a CAPTCHA layer on top.

**Comment by izoukhai** [comment]
@tulindor if you want an example you can find one here: https://www.izoukhai.com/blog/create-a-form-builder-with-directus-nextjs
